domain/access/state/permission.go

```diff
@@ -644,6 +648,86 @@ AND    m.cloud_credential_name = $input.cred_name
 	return results, nil
 }
 
+// ImportOfferAccess imports the user access for offers in the model.
+func (st *PermissionState) ImportOfferAccess(
+	ctx context.Context,
+	importAccess []access.OfferImportAccess,
+) error {
+	db, err := st.DB(ctx)
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	userNames := set.NewStrings()
+	for _, imp := range importAccess {
+		userNames = userNames.Union(set.NewStrings(slices.Collect(maps.Keys(imp.Access))...))
+	}
+	userNames.Add(corepermission.EveryoneUserName.String())
+
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		namesAndUUIDs, err := st.getUserUUIDs(ctx, tx, userNames.Values())
+		if err != nil {
+			return errors.Errorf("getting user uuids: %w", err)
+		}
+		perms, err := encodeOfferPermissions(namesAndUUIDs, importAccess)
+		if err != nil {
+			return errors.Errorf("encoding offer permissions: %w", err)
+		}
+		for _, perm := range perms {
+			if err = insertPermission(ctx, tx, perm); err != nil {
+				return errors.Capture(err)
+			}
+		}
+		return nil
+	})
+
+	return errors.Capture(err)
+}
+
+func encodeOfferPermissions(namesAndUUIDs map[string]string, input []access.OfferImportAccess) ([]dbPermission, error) {
+	var result []dbPermission
+	everyoneUUID := namesAndUUIDs[corepermission.EveryoneUserName.String()]
```

Comment:
Is this user guaranteed to exist?

Reply:
Yes, it is added by the bootstrap worker.